### PR TITLE
Robot Life() is the Life() for me

### PR DESCRIPTION
**Pull request by jknpj**
Tweaked, fixed and moved a few things around robot's life.dm, backporting a few features from MoMMI code.
Nuked MoMMI's life.dm. Almost everything there was already handled in robot code and i have no clue how the fuck this worked without making mommi gas. Also how the fuck no one noticed the duplicated clockwork mommi file?

:cl:
 * bugfix: Fixed robots jittering forever
 * bugfix: Fixed a few issues with clockwork MoMMIs

### CODE REVIEW

File: code/modules/mob/living/silicon/mommi/mommi_inventory.dm [hunk 1]
```diff
@@ -330,3 +328,18 @@
 			update_items()
 		else
 			to_chat(M, "<span class='warning'>You are unable to equip that.</span>")
+
+// MoMMIs only have one hand.
+/mob/living/silicon/robot/mommi/update_items()
+	if(client)
+		client.screen -= contents
+		for(var/obj/I in src.contents)
+			if(I)
```
Comment (MadmanMartian) [inline code/modules/mob/living/silicon/mommi/mommi_inventory.dm]: How would this ever be null when you're iterating over each object?

File: code/modules/mob/living/silicon/robot/life.dm [hunk 2]
```diff
@@ -37,16 +44,16 @@
...
+			cell.use(1)
 		else
-			if(module_state_1)
-				cell.use(3)
-			if(module_state_2)
-				cell.use(3)
-			if(module_state_3)
-				cell.use(3)
+			for(var/M in get_all_slots())
+				if(M)
```
Comment (MadmanMartian) [inline code/modules/mob/living/silicon/robot/life.dm]: Again, how woukd this be null?

File: code/modules/mob/living/silicon/robot/robot_hud.dm [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+				clear_alert(SCREEN_ALARM_TEMPERATURE)
+				return FALSE
+	else //vacuums are cold
+		throw_alert(SCREEN_ALARM_TEMPERATURE, /obj/abstract/screen/alert/robot/temp/cold, -2)
+
+/mob/living/silicon/robot/proc/handle_health_hud()
+	if(healths)
+		if(!isDead())
+			switch(health)
+				if(200 to INFINITY)
```
Comment (MadmanMartian) [inline code/modules/mob/living/silicon/robot/robot_hud.dm]: Fairly sure mommis dont have 200hp

Shouldnt this be max_health, and go down by a fifth each?

File: code/modules/mob/living/silicon/robot/robot_inventory.dm [hunk 4]
```diff
@@ -320,4 +318,18 @@
...
 /mob/living/silicon/robot/get_inactive_hand(var/obj/item/W)
-	return FALSE
\ No newline at end of file
+	return FALSE
+
+/mob/living/silicon/robot/proc/update_items()
+	if(client)
+		client.screen -= contents
+		for(var/obj/I in contents)
+			if(I && !(istype(I,/obj/item/weapon/cell) || istype(I,/obj/item/device/radio)  || istype(I,/obj/machinery/camera) || istype(I,/obj/item/device/mmi)))
```
Comment (MadmanMartian) [inline code/modules/mob/living/silicon/robot/robot_inventory.dm]: How. Would. I. Be. Null?

File: code/modules/mob/living/silicon/robot/life.dm [hunk 5]
```diff
@@ -37,16 +45,16 @@
 	adjustFireLoss(0)
 
 /mob/living/silicon/robot/proc/use_power()
-	if(is_component_functioning("power cell") && cell)
+	if(cell && is_component_functioning("power cell"))
```
Comment (MadmanMartian) [inline code/modules/mob/living/silicon/robot/life.dm:48]: Shouldn't it be get_cell() ?

Comment (jknpj) [reply code/modules/mob/living/silicon/robot/life.dm:48]: cell is already in its typecast, if robot's get_cell() did anything other than returning cell, maybe.

File: code/modules/mob/living/silicon/robot/robot_hud.dm [hunk 6]
```diff
@@ -0,0 +1,132 @@
+/mob/living/silicon/robot/handle_regular_hud_updates()
```
Comment (Kurfursten) [inline code/modules/mob/living/silicon/robot/robot_hud.dm:1]: couldn't we move much of this file up to a living level?, e.g.: processing huds, blindness, blurry, etc.

Comment (jknpj) [reply code/modules/mob/living/silicon/robot/robot_hud.dm:1]: Maybe, but should that really be bundled in the same PR that nuked mommi's life.dm?

**Pull request merged**